agent/bench-scripts/postprocess/uperf-postprocess

```diff
@@ -192,55 +204,99 @@ foreach $result_file (sort readdir($dh) ) {
+				$prev{'ops'} = $ops;
+				$prev{'bytes'} = $bytes;
 			}
 		}
 		$nr_result_files++;
 		close(IN_FILE);
-		# the uid should be unique
-		if ( (@latency_samples) ) {
-			trim_series(\@latency_samples, $series_trim, $series_trim);
+
+		# Trim the last several seconds of data from the series.
+		my $limit = $prev{'timestamp'} - $series_trim * 1000;
+		foreach my $ts (sort {$b <=> $a} keys %latency_samples) {  # Reverse-order
+			if ($ts <= $limit) {
+				last;
+			}
+			delete $latency_samples{$ts};
+		}
+
+		if (%latency_samples) {
 			my %this_lat_dataset;
 			my $description = "Average elapsed time spanning: client sending, server accepting/sending, and client receiving 1 message, over a 1 second window";
-			my $mean = get_mean(\@latency_samples);
-			%this_lat_dataset = ( 	get_label('uid_label') => create_uid('client_hostname_label', 'server_hostname_label', 'server_port_label'), get_label('description_label') => $description,
-						get_label('role_label') => $role, get_label('client_hostname_label') => $client_hostname,
-						get_label('server_hostname_label') => $server_hostname, get_label('server_port_label') => "$server_port",
-						get_label('value_label') => $mean, get_label('timeseries_label') => \@latency_samples   );
+			my $mean = get_mean_hash(\%latency_samples);
+			%this_lat_dataset = (
+				get_label('description_label')     => $description,
+				get_label('role_label')            => $role,
+				get_label('client_hostname_label') => $client_hostname,
+				get_label('server_hostname_label') => $server_hostname,
+				get_label('server_port_label')     => "$server_port",
+				get_label('value_label')           => $mean,
+				get_label('timeseries_label')      => \%latency_samples,
+				get_label('uid_label')             => create_uid(
+					                                   'client_hostname_label',
+					                                   'server_hostname_label',
+					                                   'server_port_label')
+			);
 			push(@usec, \%this_lat_dataset);
 		}
-		if ( (@throughput_samples) ) {
-			trim_series(\@throughput_samples, $series_trim, $series_trim);
+
+		# Trim the last several seconds of data from the series.
+		foreach my $ts (sort {$b <=> $a} keys %throughput_samples) {  # Reverse-order
+			if ($ts < $limit) {
+				last;
+			}
+			delete $throughput_samples{$ts};
+		}
+
+		if (%throughput_samples) {
 			my %this_tput_dataset;
-			my $mean = get_mean(\@throughput_samples);
-			%this_tput_dataset = ( 	get_label('role_label') => $role, get_label('client_hostname_label') => $client_hostname,
-						get_label('server_hostname_label') => $server_hostname, get_label('server_port_label') => "$server_port",
-						get_label('value_label') => $mean, get_label('timeseries_label') => \@throughput_samples,
-						get_label('uid_label') => create_uid('client_hostname_label', 'server_hostname_label', 'server_port_label')  );
+			my $mean = get_mean_hash(\%throughput_samples);
+			%this_tput_dataset = (
```

Comment:
Did you want to do here what you did in linpack-postprocess and push the value directly without storing it first? Although here, it gets a bit messier because the push destination  is not fixed yet and the description field is different in the two cases.

Reply:
Nah...here it's probably worth the intermediate step.  (In either case, I believe that Perl only allocates the storage once, so it's more of a code quality/readability decision than a performance or efficiency one.)